pyface/ui/qt4/console/console_widget.py

```diff
@@ -340,6 +340,28 @@ def sizeHint(self):
     # 'ConsoleWidget' public interface
     # ---------------------------------------------------------------------------
 
+    def disconnect_event_listeners(self):
+        # Disconnect signals from __init__
+        for action in self.actions():
+            action_text = action.text()
+            if action_text == "Print":
+                action.triggered.disconnect(self.print_)
+            elif action_text == "Save as HTML/XML":
+                action.triggered.disconnect(self.export)
+            elif action_text == "Select All":
+                action.triggered.disconnect(self.select_all)
+        # Disconnect signals from _create_control
+        control = self._control
+        control.cursorPositionChanged.disconnect(self._cursor_position_changed)
+        control.customContextMenuRequested.disconnect(
+            self._custom_context_menu_requested
+        )
+        control.copyAvailable.disconnect(self.copy_available)
+        control.redoAvailable.disconnect(self.redo_available)
+        control.undoAvailable.disconnect(self.undo_available)
+        layout = control.document().documentLayout()
+        layout.documentSizeChanged.disconnect(self._adjust_scrollbars)
```

Comment:
There are quite a lot of signals, and it requires keeping some text between this method and `__init__` in sync. The `_connections_to_remove` pattern might be more maintainable and easier to confirm it is right?

Reply:
I did switch to use `_connections_to_remove` but without the use of Traits. The module wasn't using anything from Traits before and I wasn't sure if it's a good idea to bring it in.